src/tbb/cgroup_info.h

```diff
@@ -0,0 +1,314 @@
+        if (!has_cgroup_v2 && !has_cgroup_v1_cpu_controller) {
+            return false; // No CPU constraints found
+        } else if (has_cgroup_v1_cpu_controller || !is_cgroup_v2_with_root_path) {
+            return true;
+        }
+        __TBB_ASSERT(!has_cgroup_v1_cpu_controller && has_cgroup_v2 && is_cgroup_v2_with_root_path, nullptr);
+
+        // Usually PID 1 is the init process
+        unique_file_t init_process_cgroup_file(std::fopen("/proc/1/cgroup", "r"), &close_file);
+        cgroup_paths init_process_cgroup_cache{};
+        cache_relative_path_for(init_process_cgroup_file.get(), init_process_cgroup_cache);
+        if (*init_process_cgroup_cache.v2_relative_path) {
+            // If the init process cgroup path is "/init.scope", it means systemd is used
+            if (!std::strncmp(init_process_cgroup_cache.v2_relative_path, "/init.scope", 11)) {
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    static int try_common_cgroup_mount_path(const cgroup_paths& relative_paths_cache) {
+        int num_cpus = error_value;
+        char dir[PATH_MAX] = {0};
+        if (*relative_paths_cache.v2_relative_path) {
+            if (std::snprintf(dir, PATH_MAX, "%s/%s", "/sys/fs/cgroup", relative_paths_cache.v2_relative_path) >= 0) {
+                try_read_cgroup_v2_num_cpus_from(dir, num_cpus);
+            }
+        }
+
+        if (num_cpus == error_value && *relative_paths_cache.v1_relative_path) {
+            if (std::snprintf(dir, PATH_MAX, "%s/%s", "/sys/fs/cgroup", relative_paths_cache.v1_relative_path) >= 0) {
+                try_read_cgroup_v1_num_cpus_from(dir, num_cpus);
+            }
+        }
+        return num_cpus;
+    }
+
+    static int parse_cpu_constraints() {
+        // Reading /proc/self/cgroup anyway, so open it right away
+        unique_file_t cgroup_file_ptr(std::fopen("/proc/self/cgroup", "r"), &close_file);
+        if (!cgroup_file_ptr)
+            return error_value; // Failed to open cgroup file
+
+        cgroup_paths relative_paths_cache;
+        if (!is_cpu_restriction_possible(cgroup_file_ptr.get(), relative_paths_cache)) {
+            return unlimited_num_cpus;
+        }
+
+        int found_num_cpus = error_value; // Initialize to an impossible value
+        found_num_cpus = try_common_cgroup_mount_path(relative_paths_cache);
+        if (found_num_cpus != error_value) {
+            return found_num_cpus;
+        }
+
+        auto close_mounts_file = [](std::FILE* file) { endmntent(file); };
+        using unique_mounts_file_t = std::unique_ptr<std::FILE, decltype(close_mounts_file)>;
+        unique_mounts_file_t mounts_file_ptr(setmntent("/proc/self/mounts", "r"), close_mounts_file);
+        if (!mounts_file_ptr)
+            return error_value;
+
+        // TODO: To avoid parsing /proc/self/mounts, read relative paths first and try opening
+        //       necessary files in the most probable mount points (see RFC for details)
```

Comment:
Is not that TODO already done in `try_common_cgroup_mount_path`?

Reply:
Removed.